compiler/z/codegen/OMRRegisterDependencyStruct.hpp

```diff
@@ -42,59 +42,16 @@ namespace OMR { typedef OMR::Z::RegisterDependencyExt RegisterDependency; }
 #include "infra/Array.hpp"            // for TR_Array
 #include "infra/Assert.hpp"           // for TR_ASSERT
 
-namespace TR { struct RegisterDependency; }
-
 namespace OMR
 {
 namespace Z
 {
-struct RegisterDependencyExt
+struct RegisterDependencyExt : OMR::RegisterDependencyExt
    {
-   uint32_t                   _realRegister:7;
-   uint32_t                   _flags:2;
-   uint32_t                   _virtualRegIndex:23;
-
-   #define   BAD_VIRTUAL_IDX  0x7FFFFF
-   void operator=(const TR::RegisterDependency &other);
-
-   TR::RealRegister::RegNum getRealRegister() {return (TR::RealRegister::RegNum) _realRegister;}
-   void setRealRegister(TR::RealRegister::RegNum r)
-      {
-      _realRegister = (uint32_t) r;
-      }
-
-   uint32_t getFlags()             {return _flags;}
-   uint32_t assignFlags(uint8_t f) {return _flags = f;}
-   uint32_t setFlags(uint8_t f)    {return _flags |= f;}
-   uint32_t resetFlags(uint8_t f)  {return _flags &= ~f;}
-
-   uint32_t getDefsRegister()   {return _flags & DefinesDependentRegister;}
-   uint32_t setDefsRegister()   {return _flags |= DefinesDependentRegister;}
-   uint32_t resetDefsRegister() {return _flags &= ~DefinesDependentRegister;}
-
-   uint32_t getRefsRegister()   {return _flags & ReferencesDependentRegister;}
-   uint32_t setRefsRegister()   {return _flags |= ReferencesDependentRegister;}
-   uint32_t resetRefsRegister() {return _flags &= ~ReferencesDependentRegister;}
-
-   TR::Register * getRegister(TR::CodeGenerator *cg)
-      {
-       if (_virtualRegIndex==BAD_VIRTUAL_IDX)
-          return NULL;
-       else
-          return (cg->getRegisterArray())[_virtualRegIndex];
-      }
+   TR::RealRegister::RegNum  _realRegister;
```

Comment:
@0xdaryl note this can and should be commoned however I think x86 is blocking this at the moment because we do not use `RegNum` (last I looked) in the x86 codegen.